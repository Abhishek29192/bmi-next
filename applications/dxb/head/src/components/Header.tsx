import Button from "@bmi-digital/components/button";
import { useIsClient } from "@bmi-digital/components/hooks";
import HeaderComponent from "@bmi-digital/components/header";
import HidePrint from "@bmi-digital/components/hide-print";
import IconButton from "@bmi-digital/components/icon-button";
import ArrowForwardIcon from "@bmi-digital/components/icon/ArrowForward";
import { RegionCode } from "@bmi-digital/components/language-selection";
import { microCopy } from "@bmi/microcopies";
import { graphql, Link, withPrefix } from "gatsby";
import React, { useMemo } from "react";
import AuthService from "../auth/service";
import Image from "../components/Image";
import { useConfig } from "../contexts/ConfigProvider";
import { useBasketContext } from "../contexts/SampleBasketContext";
import useAuth from "../hooks/useAuth";
import { constructUrlWithPrevPage } from "../templates/myAccountPage/utils";
import { checkIfActiveLabelInParentNode } from "../utils/breadcrumbUtils";
import { pushToDataLayer, useGTM } from "../utils/google-tag-manager";
import { getPathWithCountryCode } from "../utils/path";
import { stringifyToObject } from "../utils/createActionLabelForAnalytics";
import Icon from "./Icon";
import RichText from "./RichText";
import SampleBasketDialog from "./SampleBasketDialog";
import { Data as SiteData, useSiteContext } from "./Site";
import { getCTA } from "./link/utils";
import type { RichTextData } from "./RichText";
import type { IconButtonProps } from "@bmi-digital/components/icon-button";
import type { ButtonProps } from "@bmi-digital/components/button";
import type { Data as PageInfoData } from "./PageInfo";
import type {
  Data as LinkData,
  NavigationData,
  NavigationItem
} from "./link/types";
import type { GetMicroCopy } from "./MicroCopy";
import type { NavigationList } from "@bmi-digital/components/navigation";
import type { Data as PromoData } from "./Promo";

type ParseNavigationProps = {
  navigationItems: (NavigationData | NavigationItem | LinkData)[];
  countryCode: string;
  getMicroCopy: GetMicroCopy;
  isClient?: boolean;
};

type GetPromoSectionProps = {
  promo: PromoData;
  countryCode: SiteData["countryCode"];
  getMicroCopy: GetMicroCopy;
};

const getPromoSection = (props: GetPromoSectionProps) => {
  const { promo, countryCode, getMicroCopy } = props;
  const cta = getCTA(
    promo,
    countryCode,
    getMicroCopy(microCopy.PAGE_LINK_LABEL)
  );

  return [
    {
      label: promo.title || promo.name,
      image: promo.featuredMedia ? (
        <Image {...promo.featuredMedia} />
      ) : undefined
    },
    { label: promo.title || promo.name, isHeading: true },
    ...(promo.subtitle ? [{ label: promo.subtitle, isParagraph: true }] : []),
    ...(cta ? [{ label: promo.cta.label || promo.name, action: cta }] : [])
  ];
};

const parseNavigation = ({
  navigationItems,
  countryCode,
  getMicroCopy,
  isClient
}: ParseNavigationProps): NavigationList[] => {
  if (!navigationItems || navigationItems.length === 0) {
    return [];
  }

  return navigationItems.reduce((result, { __typename, ...item }) => {
    if (__typename === "ContentfulNavigation") {
      const { label, links, link, promos } = item as NavigationData;

      const navItem = {
        label,
        menu: parseNavigation({
          navigationItems: link ? [link, ...links] : links,
          countryCode,
          getMicroCopy,
          isClient
        }),
        ...(promos &&
          promos.length && {
            footer: promos.flatMap((promo) =>
              getPromoSection({ promo, countryCode, getMicroCopy })
            )
          }),
        gtm: {
          id: "nav-main-menu",
          label: label
        }
      };

      return result.concat(navItem);
    }

    if (__typename === "ContentfulNavigationItem") {
      const { value, type } = item as NavigationItem;

      if (type === "Heading") {
        return result.concat({
          label: value,
          isLabelHidden: false,
          isHeading: true
        });
      }

      if (type === "Separator") {
        if (result.length === 0) {
          return result;
        }
        const lastItem = result[result.length - 1];
        result[result.length - 1] = { ...lastItem, hasSeparator: true };
        return result;
      }
    }

    if (__typename === "ContentfulLink") {
      let action;

      const {
        label,
        isLabelHidden,
        icon: iconName,
        linkedPage,
        url
      } = item as LinkData;

      if (linkedPage) {
        action = {
          to: getPathWithCountryCode(countryCode, linkedPage.path),
          component: Link
        };
      } else if (url) {
        const href =
          isClient && url.includes(process.env.GATSBY_INTOUCH_ORIGIN)
            ? constructUrlWithPrevPage(url)
            : url;
        action = {
          href
        };
      }

      return result.concat({
        label,
        ...(iconName && { icon: <Icon name={iconName} /> }),
        isLabelHidden,
        action,
        gtm: {
          id: "nav-main-menu",
          label: label,
          action: stringifyToObject(action?.to) || action?.href
        }
      });
    }

    return result;
  }, []);
};

const onCountrySelection = (label: string, code: string) =>
  pushToDataLayer({ id: "nav-country-selector", label: label, action: code });

export type Region = {
  label: string;
  regionCode: RegionCode;
  menu: Array<{
    code: string;
    label: string;
    icon: string | null;
  }>;
};

const Header = ({
  navigationData,
  utilitiesData,
  countryCode,
  activeLabel,
  countryNavigationIntroduction,
  regions,
  sampleBasketLink,
  maximumSamples,
  lastNavigationLabel,
  disableSearch
}: {
  navigationData: NavigationData | null;
  utilitiesData: NavigationData | null;
  countryCode: string;
  activeLabel?: string;
  countryNavigationIntroduction?: RichTextData | null;
  regions: Region[];
  sampleBasketLink?: PageInfoData;
  maximumSamples: number | null;
  lastNavigationLabel?: string;
  disableSearch?: boolean;
}) => {
  const { isClient } = useIsClient();

  const languages = useMemo(
    () =>
      regions.map((region) => ({
        ...region,
        menu: region.menu.map((language) => ({
          ...language,
          icon: language.icon ? withPrefix(language.icon) : undefined
        }))
      })),
    [regions]
  );

  const language = useMemo(
    () =>
      languages
        .reduce<(typeof languages)[0]["menu"]>(
          (acc, { menu }) => acc.concat(menu),
          []
        )
        .find((l) => l.code === countryCode),
    [languages, countryCode]
  );

  const { getMicroCopy, accountPage } = useSiteContext();
  const {
    isGatsbyDisabledElasticSearch,
    isSampleOrderingEnabled,
    isLoginEnabled
  } = useConfig();
  const {
    basketState: { products: productsInBasket }
  } = useBasketContext();

  const { isLoggedIn, profile } = useAuth();

  if (!navigationData || !utilitiesData) {
    return null;
  }

  const utilities = parseNavigation({
    navigationItems: utilitiesData.links,
    countryCode,
    getMicroCopy,
    isClient
  });

  if (isLoginEnabled) {
    if (isLoggedIn && profile) {
      utilities.push(
        {
          label: getMicroCopy(microCopy.MY_ACCOUNT_LABEL),
          action: {
            to: getPathWithCountryCode(countryCode, accountPage?.slug),
            component: Link,
            "data-testid": "my-acc"
          }
        },
        {
          label: getMicroCopy(microCopy.LOG_OUT_LABEL_BTN),
          action: {
            onClick: () => {
              AuthService.logout();
            },
            "data-testid": "logout"
          }
        }
      );
    } else {
      utilities.push({
        label: getMicroCopy(microCopy.LOG_IN_LABEL_BTN),
        action: {
          onClick: () => {
            AuthService.login();
          },
          "data-testid": "login"
        }
      });
    }
  }

  const navigation = parseNavigation({
    navigationItems: navigationData.links,
    countryCode,
    getMicroCopy,
    isClient
  });

  const basketUrl =
    sampleBasketLink &&
    getPathWithCountryCode(countryCode, sampleBasketLink.slug);

  const newActiveLabel =
    checkIfActiveLabelInParentNode(lastNavigationLabel, navigationData) ||
    activeLabel;
  return (
    <HidePrint
      component={() => (
        <HeaderComponent
          disableSearch={isGatsbyDisabledElasticSearch || disableSearch}
          languages={languages}
          language={language}
          languageLabel={getMicroCopy(microCopy.MENU_LANGUAGE)}
          languageIntroduction={
            <RichText
              hasNoBottomMargin
              document={countryNavigationIntroduction}
            />
          }
          utilities={utilities}
          navigation={navigation}
          logoAction={{
            component: Link,
            to: getPathWithCountryCode(countryCode, "")
          }}
          logoLabel={getMicroCopy(microCopy.GLOBAL_LOGO_LABEL)}
          activeNavLabel={newActiveLabel}
          closeLabel={getMicroCopy(microCopy.GLOBAL_CLOSE)}
          searchGtm={{
            id: "search1",
            label: getMicroCopy(microCopy.SEARCH_LABEL)
          }}
          navUtilityLinkButton={(props: ButtonProps) => (
            <Button
              gtm={{
                id: "nav-top-utilities-bar",
                action: stringifyToObject(props.to) || props.href,
                label: React.Children.toArray(props.children).join("")
              }}
              {...props}
            />
          )}
          shoppingCartCount={productsInBasket.length}
          basketLabel={getMicroCopy(microCopy.BASKET_LABEL)}
          sampleBasketDialog={
            isSampleOrderingEnabled && sampleBasketLink?.sections?.[0]?.title
              ? (props: { toggleCart: () => void }) => (
                  <SampleBasketDialog
                    title={sampleBasketLink.sections[0].title}
                    basketUrl={basketUrl}
                    maximumSamples={maximumSamples}
                    {...props}
                  />
                )
              : null
          }
          navigationButtonComponent={(props: ButtonProps) => (
            <Button
              gtm={{
                id: "nav-main-menu",
                action: stringifyToObject(props.to),
                label:
                  typeof props.children !== "string"
                    ? props.accessibilityLabel
                    : undefined
              }}
              {...props}
            />
          )}
          promoButtonComponent={({ startIcon, ...props }: ButtonProps) => (
            <Button
              {...props}
              variant="outlined"
              endIcon={<ArrowForwardIcon />}
              className="Button"
              style={{ marginLeft: 16, marginBottom: 15 }}
            />
          )}
          closeButtonComponent={(props: IconButtonProps) => (
            <IconButton
              gtm={{
                id: "nav-country-selector",
                label: "close panel",
                action: "close panel"
              }}
              {...props}
            />
          )}
          onCountrySelection={onCountrySelection}
          useGTM={useGTM}
          searchAction={getPathWithCountryCode(countryCode, "search")}
          searchLabel={getMicroCopy(microCopy.SEARCH_LABEL)}
          searchPlaceholder={getMicroCopy(microCopy.SEARCH_PLACEHOLDER_HEADER)}
          searchTitle={getMicroCopy(microCopy.SEARCH_TITLE)}
          openLabel={getMicroCopy(microCopy.MENU_OPEN)}
          mainMenuTitleLabel={getMicroCopy(microCopy.MENU_MAIN_TITLE)}
          mainMenuDefaultLabel={getMicroCopy(microCopy.MENU_MAIN_DEFAULT)}
        />
      )}
    />
  );
};

export default Header;

export const query = graphql`
  fragment HeaderNavigationFragment on ContentfulNavigation {
    __typename
    label
    links {
      __typename
      ... on ContentfulLink {
        ...LinkFragment
      }
      ... on ContentfulNavigation {
        link {
          ...LinkFragment
        }
        label
        promos {
          ... on ContentfulPromoOrPage {
            __typename
            ...PromoHeaderFragment
            ...PageInfoHeaderFragment
          }
        }
        links {
          __typename
          ... on ContentfulNavigationItem {
            type
            value
          }
          ... on ContentfulLink {
            ...LinkFragment
          }
          ... on ContentfulNavigation {
            link {
              ...LinkFragment
            }
            label
            links {
              __typename
              ... on ContentfulNavigationItem {
                type
                value
              }
              ... on ContentfulLink {
                ...LinkFragment
              }
              ... on ContentfulNavigation {
                link {
                  ...LinkFragment
                }
                label
                links {
                  __typename
                  ... on ContentfulNavigationItem {
                    type
                    value
                  }
                  ... on ContentfulLink {
                    ...LinkFragment
                  }
                  ... on ContentfulNavigation {
                    link {
                      ...LinkFragment
                    }
                    label
                    links {
                      __typename
                      ... on ContentfulNavigationItem {
                        type
                        value
                      }
                      ... on ContentfulLink {
                        ...LinkFragment
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  }
  fragment HeaderUtilitiesFragment on ContentfulNavigation {
    label
    links {
      __typename
      ... on ContentfulLink {
        ...LinkFragment
      }
    }
  }
  fragment HeaderLanguageFragment on ContentfulResources {
    countryNavigationIntroduction {
      ...RichTextFragment
    }
  }
  fragment RegionFragment on RegionJson {
    label
    regionCode
    menu {
      code
      label
      icon
    }
  }
`;
