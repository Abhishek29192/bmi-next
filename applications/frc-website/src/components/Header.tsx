import React from "react";
import { Icopal as IcopalLogoSVG } from "@bmi/logo";
import Paper from "@material-ui/core/Paper";
import Container from "@bmi/container";
import Typography from "@bmi/typography";
import { Link } from "gatsby";
import styles from "./Header.module.scss";

const BmiHeader = () => {
  return (
    <Paper className={styles.Header} component="header" elevation={3} square>
      <Container className={styles.container}>
        <Link
          to={
            "/" + (typeof window !== "undefined" ? window.location.search : "")
          }
        >
          <IcopalLogoSVG className={styles.logo} />
        </Link>
        <Typography variant="h5" className={styles.text}>
          Bitumen Flat Roof Estimator
        </Typography>
      </Container>
    </Paper>
  );
};

export default BmiHeader;
