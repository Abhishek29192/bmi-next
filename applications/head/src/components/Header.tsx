import React from "react";
import { graphql, Link } from "gatsby";
import AeroDek from "@bmi/logo/svgs/AeroDek.svg";
import Arrow from "@bmi/header/svgs/Arrow.svg";
import Icopal from "@bmi/logo/svgs/Icopal.svg";
import Monier from "@bmi/logo/svgs/Monier.svg";
import Monarplan from "@bmi/logo/svgs/Monarplan.svg";
import Zanda from "@bmi/logo/svgs/Zanda.svg";
import { NavigationData } from "../templates/types";
import HeaderComponent from "@bmi/header";

const IconMap = {
  AeroDek: <AeroDek />,
  Monier: <Monier />,
  Monarplan: <Monarplan />,
  Arrow: <Arrow />,
  Zanda: <Zanda />,
  Icopal: <Icopal />
};

const parseNavigation = (navigationItems) => {
  return navigationItems.reduce(
    (
      result,
      {
        label,
        links,
        icon: iconName,
        isLabelHidden,
        linkedPage,
        url,
        type,
        __typename,
        value
      }
    ) => {
      let iconLabel;
      if (isLabelHidden && iconName) {
        iconLabel = IconMap[iconName] || <Arrow />;
      }
      if (!isLabelHidden && iconName) {
        iconLabel = [IconMap[iconName] || <Arrow />, label];
      }

      const isHeading =
        __typename === "ContentfulNavigationItem" && type === "Heading";
      const isSeparator =
        __typename === "ContentfulNavigationItem" && type === "Separator";

      if (isSeparator) {
        if (result.length === 0) {
          return result;
        }
        let lastItem = result[result.length - 1];
        result[result.length - 1] = { ...lastItem, hasSeparator: true };
        return result;
      }

      let action;
      if (linkedPage) {
        action = {
          model: "routerLink",
          // TODO: use countryCode from context instead of /no
          to: `/no/${linkedPage.slug}`,
          linkComponent: Link
        };
      } else if (url) {
        action = {
          model: "htmlLink",
          href: url
        };
      }

      return result.concat({
        label: iconLabel || label || value,
        isLabelHidden,
        action,
        isHeading,
        menu: links ? parseNavigation(links) : undefined
      });
    },
    []
  );
};

const Header = ({
  navigationData,
  utilitiesData
}: {
  navigationData: NavigationData;
  utilitiesData: NavigationData;
}) => {
  if (!navigationData || !utilitiesData) {
    return null;
  }
  const utilities = parseNavigation(utilitiesData.links);
  const navigation = parseNavigation(navigationData.links);

  return <HeaderComponent utilities={utilities} navigation={navigation} />;
};

export default Header;

export const query = graphql`
  fragment HeaderNavigationFragment on ContentfulNavigation {
    label
    links {
      ... on ContentfulLink {
        ...LinkFragment
      }
      ... on ContentfulNavigation {
        label
        links {
          ... on ContentfulNavigationItem {
            type
            value
          }
          ... on ContentfulLink {
            ...LinkFragment
          }
          ... on ContentfulNavigation {
            label
            links {
              ... on ContentfulNavigationItem {
                type
                value
              }
              ... on ContentfulLink {
                ...LinkFragment
              }
              ... on ContentfulNavigation {
                label
                links {
                  ... on ContentfulLink {
                    ...LinkFragment
                  }
                  ... on ContentfulNavigation {
                    label
                    links {
                      ... on ContentfulLink {
                        ...LinkFragment
                      }
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
  }
  fragment HeaderUtilitiesFragment on ContentfulNavigation {
    label
    links {
      ... on ContentfulLink {
        ...LinkFragment
      }
    }
  }
`;
