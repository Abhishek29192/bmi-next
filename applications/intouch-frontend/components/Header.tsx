import React from "react";
import Typography from "@bmi/typography";
import styles from "./styles/Header.module.scss";

interface Props {
  title: string;
}

export const Header = ({ title }: Props) => {
  return (
    <div className={styles.header}>
      <Typography variant="h3">{title}</Typography>
    </div>
  );
};
