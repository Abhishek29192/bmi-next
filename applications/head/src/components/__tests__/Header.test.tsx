import React from "react";
import Header from "../Header";
import { render } from "@testing-library/react";
import mockConsole from "jest-mock-console";

beforeAll(() => {
  mockConsole();
});

describe("Header component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <Header
        navigationData={{
          label: "Main",
          links: [
            {
              label: "Get in touch",
              links: [
                {
                  id: "",
                  label: "+44 (0) 1234567890",
                  url: "tel:+4401234567890",
                  isLabelHidden: null,
                  icon: "Phone",
                  linkedPage: null
                }
              ]
            },
            {
              label: "About BMI",
              links: [
                {
                  id: "",
                  label: "Our story",
                  url: null,
                  isLabelHidden: null,
                  icon: null,
                  linkedPage: {
                    slug: "landing-page"
                  }
                }
              ]
            }
          ]
        }}
        utilitiesData={{
          label: "About BMI",
          links: [
            {
              id: "",
              label: "Our story",
              url: null,
              isLabelHidden: null,
              icon: null,
              linkedPage: {
                slug: "landing-page"
              }
            }
          ]
        }}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
