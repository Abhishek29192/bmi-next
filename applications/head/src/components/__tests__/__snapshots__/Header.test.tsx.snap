// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
<header
  class="MuiPaper-root Header MuiPaper-elevation3"
>
  <nav
    aria-label="Utilities"
    class="UtilitiesBar"
    role="navigation"
  >
    <div
      class="MuiContainer-root Container MuiContainer-maxWidthXl"
    >
      <ul>
        <li
          class="NavItem"
        >
          <a
            class="MuiButtonBase-root MuiButton-root MuiButton-text Button UtilitiesButton MuiButton-textPrimary"
            href="/gb/landing-page"
          >
            <span
              class="MuiButton-label"
            >
              Our story
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    class="NavigationBar"
  >
    <div
      class="MuiContainer-root Container MuiContainer-maxWidthXl"
    >
      <div
        class="NavigationBar__Left"
      >
        <a
          class="LogoLink"
          href="/gb/"
        >
          <svg
            class="Logo"
          />
        </a>
        <nav
          aria-label="Navigation"
          class="Navigation"
          role="navigation"
        >
          <div
            class="MuiTabs-root"
          >
            <div
              class="MuiTabs-scrollable"
              style="width: 99px; height: 99px; position: absolute; top: -9999px; overflow: scroll;"
            />
            <div
              class="MuiTabs-scroller MuiTabs-scrollable"
              style="margin-bottom: 0px;"
            >
              <div
                class="MuiTabs-flexContainer"
                role="tablist"
              >
                <button
                  aria-controls="navigation-tabpanel-0"
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit NavItem MuiTab-labelIcon"
                  id="navigation-tab-0"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                      />
                    </svg>
                    Get in touch
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
                <button
                  aria-controls="navigation-tabpanel-1"
                  aria-selected="false"
                  class="MuiButtonBase-root MuiTab-root MuiTab-textColorInherit NavItem MuiTab-labelIcon"
                  id="navigation-tab-1"
                  role="tab"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiTab-wrapper"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                      />
                    </svg>
                    About BMI
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <span
                class="PrivateTabIndicator-root-1 PrivateTabIndicator-colorSecondary-3 MuiTabs-indicator"
                style="left: 0px; width: 0px;"
              />
            </div>
          </div>
        </nav>
      </div>
      <div
        class="NavigationBar__Right"
      >
        <button
          aria-label="Search"
          class="MuiButtonBase-root MuiIconButton-root IconButton IconButton--medium IconButton--contained"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          aria-label="Open menu"
          class="MuiButtonBase-root MuiIconButton-root IconButton IconButton--medium IconButton--contained BurgerButton"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root Backdrop"
    style="opacity: 0; visibility: hidden;"
  />
  <div
    class="Drawer NavDrawer"
    style="visibility: hidden; webkit-transform: translateY(-NaNpx); transform: translateY(-NaNpx);"
  >
    <button
      aria-label="Close"
      class="MuiButtonBase-root MuiIconButton-root IconButton IconButton--medium IconButton--contained CloseButton"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <nav
      class="Navigation"
    >
      <div
        class="NavigationList Offset0"
      >
        <ul>
          <li>
            <h6
              class="MuiTypography-root MainMenuTitle Typography MuiTypography-h6"
            >
              BMI Group
            </h6>
          </li>
          <li>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text Button NavigationListButton MuiButton-textPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Get in touch
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                    />
                  </svg>
                </span>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <div
              class="NavigationList"
            >
              <ul>
                <li
                  class="BackNavigation"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text Button NavigationListButton BackButton MuiButton-textPrimary"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <span
                        class="MuiButton-startIcon MuiButton-iconSizeMedium"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                          />
                        </svg>
                      </span>
                      Main menu
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <hr
                    class="Separator"
                  />
                </li>
                <li>
                  <a
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiButton-root MuiButton-text Button NavigationListButton MuiButton-textPrimary"
                    href="tel:+4401234567890"
                    tabindex="0"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M6.62 10.79c1.44 2.83 3.76 5.14 6.59 6.59l2.2-2.2c.27-.27.67-.36 1.02-.24 1.12.37 2.33.57 3.57.57.55 0 1 .45 1 1V20c0 .55-.45 1-1 1-9.39 0-17-7.61-17-17 0-.55.45-1 1-1h3.5c.55 0 1 .45 1 1 0 1.25.2 2.45.57 3.57.11.35.03.74-.25 1.02l-2.2 2.2z"
                        />
                      </svg>
                      <b>
                        +44 (0) 1234567890
                      </b>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </a>
                </li>
              </ul>
            </div>
          </li>
          <li>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text Button NavigationListButton MuiButton-textPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                About BMI
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                    />
                  </svg>
                </span>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <div
              class="NavigationList"
            >
              <ul>
                <li
                  class="BackNavigation"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-text Button NavigationListButton BackButton MuiButton-textPrimary"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      <span
                        class="MuiButton-startIcon MuiButton-iconSizeMedium"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                          />
                        </svg>
                      </span>
                      Main menu
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                  <hr
                    class="Separator"
                  />
                </li>
                <li>
                  <a
                    class="MuiButtonBase-root MuiButton-root MuiButton-text Button NavigationListButton MuiButton-textPrimary"
                    href="/gb/landing-page"
                  >
                    <span
                      class="MuiButton-label"
                    >
                      Our story
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </a>
                </li>
              </ul>
            </div>
          </li>
          <ul
            class="Utilities"
          >
            <li>
              <a
                class="MuiButtonBase-root MuiButton-root MuiButton-text Button NavigationListButton MuiButton-textPrimary"
                href="/gb/landing-page"
              >
                <span
                  class="MuiButton-label"
                >
                  Our story
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </li>
          </ul>
          <li>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text Button NavigationListButton MuiButton-textPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Language
                <span
                  class="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                    />
                  </svg>
                </span>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </li>
        </ul>
      </div>
    </nav>
  </div>
  <div
    class="Drawer SearchDrawer"
    style="visibility: hidden; webkit-transform: translateY(-NaNpx); transform: translateY(-NaNpx);"
  >
    <button
      aria-label="Close"
      class="MuiButtonBase-root MuiIconButton-root IconButton IconButton--medium IconButton--contained CloseButton"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <h4
      class="MuiTypography-root Typography MuiTypography-h4"
    >
      How can we help you today?
    </h4>
    <div
      class="InputGroup InputGroup--locked-sm"
    >
      <div
        class="MuiFormControl-root MuiTextField-root TextField input"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
          data-shrink="false"
        >
          Search BMI...
        </label>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiFilledInput-input"
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        aria-label="Search"
        class="MuiButtonBase-root MuiIconButton-root IconButton IconButton--medium IconButton--contained button"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</header>
`;
